src/modules/hip/kernel/fog.hpp

```diff
@@ -0,0 +1,323 @@
+    uint maskIdx = ((srcStridesNCH.z) * (maskLocArrY[id_z] + id_y)) + maskLocArrX[id_z] + id_x;
+    float4 gamma_f4 = (float4)gammaTensor[id_z];
+
+    d_float8 maskAlpha_f8, maskIntensity_f8;
+    *(d_float8_s *)&maskAlpha_f8 = *(d_float8_s *)&fogAlphaMaskPtr[maskIdx];
+    *(d_float8_s *)&maskIntensity_f8 = *(d_float8_s *)&fogIntensityMaskPtr[maskIdx];
+
+    d_float24 src_f24, dst_f24;
+
+    rpp_hip_load24_pln3_and_unpack_to_float24_pln3(srcPtr + srcIdx, srcStridesNCH.y, &src_f24);
+    fog_hip_compute(srcPtr, &src_f24.f8[0], &dst_f24.f8[0], &maskAlpha_f8, &maskIntensity_f8, &gamma_f4);
+    fog_hip_compute(srcPtr, &src_f24.f8[1], &dst_f24.f8[1], &maskAlpha_f8, &maskIntensity_f8, &gamma_f4);
+    fog_hip_compute(srcPtr, &src_f24.f8[2], &dst_f24.f8[2], &maskAlpha_f8, &maskIntensity_f8, &gamma_f4);
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+RppStatus hip_exec_fog_tensor(T *srcPtr,
+                              RpptDescPtr srcDescPtr,
+                              T *dstPtr,
+                              RpptDescPtr dstDescPtr,
+                              Rpp32f *d_fogAlphaMaskPtr,
+                              Rpp32f *d_fogIntensityMaskPtr,
+                              Rpp32f *gammaTensor,
+                              Rpp32u *maskLocArrHostX,
+                              Rpp32u *maskLocArrHostY,
+                              RpptROIPtr roiTensorPtrSrc,
+                              RpptRoiType roiType,
+                              rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = (dstDescPtr->strides.hStride + 7) >> 3;
+    int globalThreads_y = dstDescPtr->h;
+    int globalThreads_z = dstDescPtr->n;
+
+    Rpp32u maskSize = FOG_MAX_WIDTH * FOG_MAX_HEIGHT;
+    Rpp32u maskSizeFloat = maskSize * sizeof(Rpp32f);
+
+    if ((srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NHWC))
+    {
+        globalThreads_x = (dstDescPtr->strides.hStride / 3 + 7) >> 3;
+        hipLaunchKernelGGL(fog_pkd_hip_tensor,
+                           dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                           dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                           0,
+                           handle.GetStream(),
+                           srcPtr,
+                           make_uint2(srcDescPtr->strides.nStride, srcDescPtr->strides.hStride),
+                           dstPtr,
+                           make_uint2(dstDescPtr->strides.nStride, dstDescPtr->strides.hStride),
+                           d_fogAlphaMaskPtr,
+                           d_fogIntensityMaskPtr,
+                           gammaTensor,
+                           maskLocArrHostX,
```

Comment:
change the variable name from maskLocArrHostX  to maskLocOffsetX for better clarity
Applicable to all such instances in this PR